pkg/domain/infra/abi/images_list.go

```diff
@@ -28,54 +30,63 @@ func (ir *ImageEngine) List(ctx context.Context, opts entities.ImageListOptions)
 
 	summaries := []*entities.ImageSummary{}
 	for _, img := range images {
-		repoDigests, err := img.RepoDigests()
-		if err != nil {
-			return nil, fmt.Errorf("getting repoDigests from image %q: %w", img.ID(), err)
-		}
+		summary, err := func() (*entities.ImageSummary, error) {
+			repoDigests, err := img.RepoDigests()
+			if err != nil {
+				return nil, fmt.Errorf("getting repoDigests from image %q: %w", img.ID(), err)
+			}
 
-		if img.ListData.IsDangling == nil { // Sanity check
-			return nil, fmt.Errorf("%w: ListData.IsDangling is nil but should not", define.ErrInternal)
-		}
-		isDangling := *img.ListData.IsDangling
-		parentID := ""
-		if img.ListData.Parent != nil {
-			parentID = img.ListData.Parent.ID()
-		}
+			if img.ListData.IsDangling == nil { // Sanity check
+				return nil, fmt.Errorf("%w: ListData.IsDangling is nil but should not", define.ErrInternal)
```

Comment:
Only if you have other changes as it's an old nit
```suggestion
				return nil, fmt.Errorf("%w: ListData.IsDangling is nil but should not be", define.ErrInternal)
```